Implement initial functionality

m1j09830 opened the pull request:
## 🗣 Description ##

This PR contains the code to create an AMI with the [Egress-Assess tool](https://github.com/tarrell13/Egress-Assess/tree/Forked) software.

## 💭 Motivation and context ##

The Egress-Assess AMI will be used for egress testing of networks during engagements.

## 🧪 Testing ##

An AMI was built, deployed, and successfully tested in a staging environment.

Tests were conducted to ensure proper connection and exchange of data between an Egress-Assess client and the server running in the staging environment.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been approved. -->

- [x] Build final pre-release AMI and test
- [x] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [x] Create a release.

m1j09830 committed:
Set the GID for the efs_users group

We want to use the same GID for the efs_users group on all instances,
to help avoid UID/GID collisions with files written to the EFS share.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

m1j09830 committed:
Replace example playbook with Egress-Assess playbook

m1j09830 committed:
Replace skeleton template text with egress-assess

m1j09830 committed:
Update skeleton README with useful information

m1j09830 committed:
Set initial version to 1.0.0-rc.1

dav3r committed:
Add cloud-init playbook

dav3r committed:
Add vnc and xfce_cool playbooks

dav3r committed:
Bump version from 1.0.0-rc.1 to 1.0.0-rc.2

dav3r committed:
Add additional SSM permissions used by VNC role

dav3r committed:
Bump version from 1.0.0-rc.2 to 1.0.0-rc.3

m1j09830 committed:
Modify Egress-Assess archive_src and pip reqs

m1j09830 committed:
Bump version from 1.0.0-rc.3 to 1.0.0-rc.4

m1j09830 committed:
Modify Egress-Assess archive_src and pip reqs

m1j09830 committed:
Bump version from 1.0.0-rc.4 to 1.0.0-rc.5

dav3r committed:
Make @m1j09830 a code owner in this repo

m1j09830 committed:
Remove skeleton text from README

m1j09830 committed:
Add a TODO comment

m1j09830 committed:
Bump version from 1.0.0-rc.5 to 1.0.0-rc.6

jsf9k commented on src/packer.pkr.hcl:
```diff
@@ -59,15 +59,15 @@ data "amazon-ami" "debian_bullseye" {
 locals { timestamp = regex_replace(timestamp(), "[- TZ:]", "") }
 
-source "amazon-ebs" "example" {
+source "amazon-ebs" "egress-assess" {
```
We prefer snake case for HCL:
```suggestion
source "amazon-ebs" "egress_assess" {
```

jsf9k commented on src/packer.pkr.hcl:
```diff
@@ -110,7 +110,7 @@ source "amazon-ebs" "example" {
 
 build {
-  sources = ["source.amazon-ebs.example"]
+  sources = ["source.amazon-ebs.egress-assess"]
```
To maintain consistency with my previous suggestion in this file:
```suggestion
  sources = ["source.amazon-ebs.egress_assess"]
```

jsf9k commented on README.md:
```diff
@@ -1,14 +1,9 @@
-[GitHub Actions](https://github.com/features/actions) configurations
-appropriate for the major languages that we use.
+This project can be used to build a Debian AMI with
+[Egress-Assess](https://github.com/FortyNorthSecurity/Egress-Assess) installed.
```
You're installing from [tarrell13/Egress-Assess](https://github.com/tarrell13/Egress-Assess), and according to #13 you intend to continue doing so, so this link should be updated:
```suggestion
[Egress-Assess](https://github.com/tarrell13/Egress-Assess) installed.
```

jsf9k reviewed (changes requested):
Looks good!  I'm only requesting a few minor changes.

m1j09830 committed:
Correct repository name for Egress-Assess

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

m1j09830 committed:
Prefer snake case for AMI source resource name

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

jsf9k reviewed (approved):
:+1:

dav3r reviewed (approved):
Great work @m1j09830 🚀

jsf9k commented:
@m1j09830 - don't forget to finalize the version before merging.

dav3r commented:
@mcdonnnj - Please review this PR even though it already has 2 dev reviews, since I helped @m1j09830 significantly and therefore my review is fairly biased.

mcdonnnj commented on src/cloud_init.yml:
```diff
@@ -0,0 +1,16 @@
+        name: "{{ item }}"
+        enabled: yes
+      loop:
+        - cloud-init-local
+        - cloud-init
+        - cloud-config
+        - cloud-final
```
We prefer to sort things like this. If the current order is required then it should be documented. I see this appears to be an inherited thing from cribbing other Packer projects so as long so we should update those projects based on the resolution of this conversation.

```suggestion
        - cloud-config
        - cloud-final
        - cloud-init
        - cloud-init-local
```

dav3r replied:
Yes, this list of items should be sorted here and elsewhere.

mcdonnnj commented on src/egress_assess.yml:
```diff
@@ -0,0 +1,17 @@
+        # TODO: Switch to master branch after Forked branch is merged
+        # See https://github.com/cisagov/egress-assess-packer/issues/13
+        archive_src: "https://github.com/tarrell13/Egress-Assess/\
+        tarball/Forked"
```
This needs to be updated in line with the resolution of https://github.com/cisagov/kali-packer/pull/141#discussion_r1089255813.

m1j09830 replied:
Made changes in `egress_assess.yml` to reflect updated repository. Please see f877fb21591423a51ef16ce07bb25397fcd1d0d3. Updated issue #13 to properly reflect the changes and close out the issue.

mcdonnnj commented on README.md:
```diff
@@ -1,14 +1,9 @@
-[pre-commit hooks](https://pre-commit.com) and
-[GitHub Actions](https://github.com/features/actions) configurations
-appropriate for the major languages that we use.
+This project can be used to build a Debian AMI with
+[Egress-Assess](https://github.com/tarrell13/Egress-Assess) installed.
```
This also needs to be updated with the resolution of https://github.com/cisagov/kali-packer/pull/141#discussion_r1089255813.

m1j09830 replied:
Made the changes to the README per the suggestion above. Please see 14c06dc77fda40c87e7bfde16ff90cf7531b6efe

m1j09830 committed:
Modify archive_src to use Auto-Egress-Assess repo

m1j09830 committed:
Update README with proper repo name

dav3r committed:
Use a non-broken version of isort

This change (and other pre-commit hook version updates) will flow in after https://github.com/cisagov/skeleton-packer/pull/217 gets merged, but since this particular hook is blocking our tests from passing, we want to upgrade now.

m1j09830 committed:
Update src/cloud_init.yml to properly sort

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

m1j09830 committed:
Bump version from 1.0.0-rc.6 to 1.0.0-rc.7

m1j09830 committed:
Merge branch 'first-commits' of github.com:cisagov/egress-assess-packer into first-commits

m1j09830 committed:
Bump version from 1.0.0-rc.7 to 1.0.0-rc.8

m1j09830 committed:
Modify version to reflect proper release

mcdonnnj commented on terraform-build-user/main.tf:
```diff
@@ -9,6 +9,13 @@ module "iam_user" {
-  ssm_parameters = ["/cyhy/dev/users", "/ssh/public_keys/*"]
-  user_name      = "build-egress-assess-packer"
+  ssm_parameters = [
+    "/cyhy/dev/users",
+    "/ssh/public_keys/*",
```
I don't believe this configuration needs access to these SSM parameters.
```suggestion
```

dav3r replied:
Ah you are right now that we have stopped including [cisagov/ansible-role-dev-ssh-access](https://github.com/cisagov/ansible-role-dev-ssh-access) in our AMIs.  We should probably remove these [from skeleton-packer also](https://github.com/cisagov/skeleton-packer/blob/078fd4c8cbaac9597a3e599f6b9f487b54a4b5b2/terraform-build-user/main.tf#L12).

mcdonnnj replied:
@dav3r Yes I have this draft PR toward that work: https://github.com/cisagov/ami-build-iam-user-tf-module/pull/53

mcdonnnj commented on src/egress_assess.yml:
```diff
@@ -0,0 +1,14 @@
+---
+- hosts: all
+  name: Install Egress-Assess tool
```
I'm fine with the playbook name remaining the same but I think this descriptor should reflect the tool being installed.

```suggestion
  name: Install Auto-Egress-Assess tool
```

mcdonnnj reviewed (changes requested):
Thanks for resolving my previous feedback. I have two more small items from re-reviewing.

m1j09830 committed:
Update descriptor to align with repository name

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

m1j09830 committed:
Remove unnecessary access to SSM parameters

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj reviewed (approved):
LGTM ✔

m1j09830 committed:
Finalize version from 1.0.0-rc.7 to 1.0.0
